### DISCUSSION TITLE
ESP32 Yellow Board as basis?

### DISCUSSION THREAD

**Discussion by Kalrkloss** [CapibaraZero/fw#41] [Ideas]
Hey, cool project.
One idea: There are those cheap ESP32 "Yellow Boards" with touch TFT all over Aliexpress and Ebay.
Would this be a good basis for CapibaraZero?
https://randomnerdtutorials.com/cheap-yellow-display-esp32-2432s028r/

**Comment by andreock** [comment]
I don't think so, they don't seems to have a lot of GPIOs pins. It's better to get something like that https://it.aliexpress.com/item/1005007181435830.html
Remember that ESP32 doesn't have USB OTG features.

Another good alternative, if you want USB OTG is this one: https://www.lilygo.cc/products/t3-s3-lr1121
They lacks of GPIO so you will need to sacrifice SX1276 or NFC or IR.

**Reply by Kalrkloss** [reply]
The Lilygo T-Embed looks nice, but also doesn't have many GPIOs broken out. https://www.lilygo.cc/products/t-embed
Nevertheless, the case looks nice, somewhat like the Flipper Zero.

**Reply by andreock** [reply]
Really cool, maybe we can use gpio multiplexer to support every module

**Reply by Kalrkloss** [reply]
Or make our own.
Maybe we can start by collecting the requirements, like display type, size, controls, battery, included communication modules, expansion bus, etc.
BTW., the ESP32 is pretty flexible when it comes to IO functions, it can route pretty much every special function to every GPIO.

I'll start:
- Compact housing resembling a somewhat bigger USB stick
- Integrated (LIION) battery and charging chircuit
- Display: (I would prefer a smaller, transflective LCD, because it's visible in full sunlight so options:)
   a) Big touch TFT
   b) small (non touch) TFT
   c) small transflective (monochrome) LCD
- Rotary encoder + maybe additional buttons
- Expansion port with 3.3V, GND and as many GPIOs as possible, 2.54 or 1.27 pitch (2.54 is better for DIY, but'll get huge). Maybe compatible to Flipper Zero?
- Micro SD card
- Integrated communication:
  - WiFi, Bluetooth (already in the ESP32)
  - LoRa 433MHz and/or 868/915MHz
  - CC1101 433MHz and/or 868/915MHz (LoRa modules can't do ASK)
  - NFC 13,56 MHz + antenna
  - rfID 125kHz + antenna
  - Infrared receiver/transmitter. Transmitter should be high power for TV-B-Gone
  - USB A host port
  - I2S codec with microphone and loudspeaker
  
Extension modules (should have I2C EEPROM for automatic identification):
- CAN bus (ESP32 already has one, so only transceiver needed)
- wideband SDR transceiver
- VHF/UHF transceiver Module (e.g. DRA818V)
- USDR compatible HF transceiver
- simple I/Q shortwave SDR receiver
- SI4732 based all-band receiver
- additional LoRa/CC1101 for multiband and relay
- JTAG/SWD debugger

**Reply by andreock** [reply]
This can be a really good idea to save space and to make board ready for usage without soldering anything. A board based on ESP32-S3 with:
- 2 USB
- a ST7789 display(maybe 1.54") 
- CC1101/SX1276(not both)
- SD card slot
- PN532(or even something better)
- RFID(maybe)
- IR
- Li-ion battery circuit 
- 5 push button like flipper

Can be a good starting point. I don't have much time to work on it but if anyone can help me, the result will be very interesting.

**Reply by Kalrkloss** [reply]
Why two USB? The ESP32-S3 has a native USB, which, by default, acts as a composite device consisting of a CDC serial port and a JTAG interface, for programming and debugging the chip.

So, even if you dedicate the serial port to host communication, you always have the JTAG left for debugging and programming.
Sure, some commercial boards have additionally a USB-to-serial chip, connected to the ESP's hardware serial port, but this isn't necessary at all, unless you reprogram the native USB to something else and still want to have an additional USB serial port.

**Comment by Kalrkloss** [comment]
Oh, and btw: You can even use two USB CDC at the same time.

idf.py menuconfig and in Component config → TinyUSB Stack → Communication Device Class (CDC) → CDC Channel Count

**Comment by Jukmarinho** [comment]
Instalei no meu esp32s3 mas não deu imagem, com um display 168x320 com SD! tentei outos display 240x240 mas tambem não deu nada! alguem poderia me ajudar
